### PR TITLE
[CURATOR-274] v2.9.0 broken EnsurePath behavior

### PR DESCRIPTION
**Pull request by Randgalt**
Previous change to create containers broke some old behavior. Old EnsurePath class only created parents once. So, introduce new EnsureContainers class to do similar but for containers. Also, make usage in a protected method so that it can be turned off by users

### CODE REVIEW

**Comment by JasonRosenberg** [comment]
should there be tests to verify this behavior?

**Comment by Randgalt** [comment]
There's already a test for ensuring path in TestPathChildrenCache

**Comment by Randgalt** [comment]
I changed my mind and added a few tests

File: curator-framework/src/main/java/org/apache/curator/framework/EnsureContainers.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.curator.framework;
+
+import org.apache.curator.utils.EnsurePath;
```
Comment (cammckenzie) [inline curator-framework/src/main/java/org/apache/curator/framework/EnsureContainers.java]: Unused import?

**Comment by cammckenzie** [comment]
Other than that minor import, looks good to me.

**Pull request merged**